tests/utils/pod.go

```diff
@@ -105,11 +106,12 @@ func FindPodByPrefix(clientSet *kubernetes.Clientset, namespace, prefix, labelSe
 			if foundPod {
 				return true, nil
 			}
-			return false, fmt.Errorf("Unable to find pod starting with prefix %s", prefix)
-		} else {
-			return false, err
 		}
+		return false, nil
 	})
+	if !foundPod {
+		return nil, fmt.Errorf("Unable to find pod starting with prefix %s", prefix)
+	}
```

Comment:
See [PollImmediate](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L244)
Potential refactor, what do you think?
```
func FindPodByPrefix(clientSet *kubernetes.Clientset, namespace, prefix, labelSelector string) (*k8sv1.Pod, error) {
	var result k8sv1.Pod
	var foundPod bool
	err := wait.PollImmediate(2*time.Second, PodCreateTime, func() (bool, error) {
		podList, err := clientSet.CoreV1().Pods(namespace).List(metav1.ListOptions{
			LabelSelector: labelSelector,
		})
		if err != nil {
                        return false, nil // continue Poll
                }
		for _, pod := range podList.Items {
			if strings.HasPrefix(pod.Name, prefix) {
				if !foundPod {
					foundPod = true
					result = pod
				} else {
					fmt.Fprintf(GinkgoWriter, "INFO: First pod name %s\n", result.Name)
					fmt.Fprintf(GinkgoWriter, "INFO: Second pod name %s\n", pod.Name)
                                        return true, fmt.Errorf("Multiple pods starting with prefix %q in namespace %q", prefix, namespace)
				}
			}
		}
                // if we get here all pods compared with at most 1 matching pod
		return true, nil
	})
	if !foundPod {
		return nil, fmt.Errorf("Unable to find pod in namespace %q starting with prefix %q", namespace, prefix)
	}
	return &result, err
}
```